R/filters.R

```diff
@@ -0,0 +1,57 @@
+#' Main function for data filtering
+#'
+#' @param .data The data to be processed. Must be the list of 2 elements:
+#' data table and metadata table.
+#' @param .method Method of filtering. Implemented methods: by.meta.
+#' @param .query Filtering query. It's a named list of filters that will be applied
+#' to data. Elements of the list are specific for selected method:
+#' - by.meta: names are metadata column headers; each value can be string
+#' (metadata value to select for the column), vector of strings (to select multiple
+#' possible values) or numeric expression in format '>#', '<#', '>=#', '<=#' where
+#' '#' is a number (integer or float, can be negative).
+#'
+#' @examples
+#' data(immdata)
+#'
+#' # Select samples with status "MS"
+#' repFilter(immdata, "by.meta", list(Status = "MS"))
+#'
+#' # Select samples from lanes "A" and "B" with age >= 15
+#' repFilter(immdata, "by.meta", list(Lane = c("A", "B"), Age = ">=15"))
+#' @export repFilter
+repFilter <- function(.data, .method, .query, ...) {
+  switch(tolower(.method),
+    by.meta = filter_by_meta(.data, .query),
+    stop("You entered the wrong method! Please, try again.")
```

Comment:
This place is far from ideal to raise error. It doesn't tell the user what has gone wrong and what should be altered for the correct work.

Reply:
It's analogous to `repClonality` and `repDiversity`, they raise error in the same place. I will add list of correct methods to this error message.